sql/analyzer/prune_columns.go

```diff
@@ -353,6 +353,11 @@ func fixRemainingFieldsIndexes(ctx *sql.Context, a *Analyzer, node sql.Node, sco
 				return n, transform.SameTree, nil
 			}
 
+			// IndexedTableAccess contains expressions in its lookupBuilder that we don't need to fix up, so skip them
+			if _, ok := n.(*plan.IndexedTableAccess); ok {
```

Comment:
i wonder if we still need this after you merge main, i tried cleaning up scalar and relational subquery analyzer lifecycles.

Reply:
I double checked this one and confirmed that this check is still needed, even with the latest changes from main.